src/plugins/console/public/application/components/refresh_interval.tsx

```diff
@@ -0,0 +1,275 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// Note: This component source code is taken from eui version 43.1.1 https://github.com/elastic/eui/blob/v43.1.1/src/components/date_picker/auto_refresh/refresh_interval.tsx
+// We should remove it from here and import it from eui, once we upgrade eui to version 43 or above.
+
+import React, { Component, ChangeEventHandler, KeyboardEventHandler } from 'react';
+import {
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormLabel,
+  EuiI18n,
+  EuiScreenReaderOnly,
+  EuiSelect,
+  EuiSwitch,
+  RelativeOption,
+  TimeUnitId,
+  TimeUnitLabel,
+  TimeUnitLabelPlural,
+  keysOf,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { Milliseconds, ApplyRefreshInterval } from '@elastic/eui/src/components/date_picker/types';
+
+export const timeUnits: { [id in TimeUnitId]: TimeUnitLabel } = {
+  s: 'second',
+  m: 'minute',
+  h: 'hour',
+  d: 'day',
+  w: 'week',
+  M: 'month',
+  y: 'year',
+};
+
+export const timeUnitsPlural: { [id in TimeUnitId]: TimeUnitLabelPlural } = {
+  s: 'seconds',
+  m: 'minutes',
+  h: 'hours',
+  d: 'days',
+  w: 'weeks',
+  M: 'months',
+  y: 'years',
+};
+
+const refreshUnitsOptions: RelativeOption[] = keysOf(timeUnits)
+  .filter((timeUnit) => timeUnit === 'h' || timeUnit === 'm' || timeUnit === 's')
+  .map((timeUnit) => ({ value: timeUnit, text: timeUnitsPlural[timeUnit] }));
+
+const MILLISECONDS_IN_SECOND = 1000;
+const MILLISECONDS_IN_MINUTE = MILLISECONDS_IN_SECOND * 60;
+const MILLISECONDS_IN_HOUR = MILLISECONDS_IN_MINUTE * 60;
+
+function fromMilliseconds(milliseconds: Milliseconds): EuiRefreshIntervalState {
+  const round = (value: number) => parseFloat(value.toFixed(2));
+  if (milliseconds > MILLISECONDS_IN_HOUR) {
+    return {
+      units: 'h',
+      value: round(milliseconds / MILLISECONDS_IN_HOUR),
+    };
+  }
+
+  if (milliseconds > MILLISECONDS_IN_MINUTE) {
+    return {
+      units: 'm',
+      value: round(milliseconds / MILLISECONDS_IN_MINUTE),
+    };
+  }
+
+  return {
+    units: 's',
+    value: round(milliseconds / MILLISECONDS_IN_SECOND),
+  };
+}
+
+function toMilliseconds(units: TimeUnitId, value: Milliseconds) {
+  switch (units) {
+    case 'h':
+      return Math.round(value * MILLISECONDS_IN_HOUR);
+    case 'm':
+      return Math.round(value * MILLISECONDS_IN_MINUTE);
+    case 's':
+    default:
+      return Math.round(value * MILLISECONDS_IN_SECOND);
+  }
+}
+
+export interface EuiRefreshIntervalProps {
+  /**
+   * Is refresh paused or running.
+   */
+  isPaused?: boolean;
+  /**
+   * Refresh interval in milliseconds.
+   */
+  refreshInterval?: Milliseconds;
+  /**
+   * Passes back the updated state of `isPaused` and `refreshInterval`.
+   */
+  onRefreshChange: ApplyRefreshInterval;
+}
+
+interface EuiRefreshIntervalState {
+  value: number | '';
+  units: TimeUnitId;
+}
+
+export class EuiRefreshInterval extends Component<
```

Comment:
I don't see why we should modify it

Reply:
The only thing, which is coming to my mind is that it can confuse somebody (the `Eui*` prefix at this component).
But still, I'm not insisting here. That is more according to your and code owners' taste )